input/pagecontent/smart-web-messaging.md

```diff
@@ -388,15 +410,255 @@ targetWindow.postMessage({
       // additional details as needed
     }
   }
-}, targetOrigin);
+}, ehrOrigin);
+```
+
+The EHR then performs the requested operation, creating an object in an internal
+scratchpad and assigning it id `123` before finally responding with:
+
+```js
+// From EHR -> app
+appWindow.postMessage({
+  "messageId": "<some new uid>",
+  "responseToMessageId": "<corresponding request messageId>",
+  "payload": {
+    "status": "201 CREATED",
+    "location": "ServiceRequest/123"
+  }
+}, appOrigin);
+```
+
+##### `scratchpad.read`
+
+###### Request `payload`
+
+| Property              | Optionality | Type   | Description |
+| --------------------- | ----------- | ------ | ----------- |
+| `location`            | OPTIONAL    | string | Takes the form `ResourceType/Id` when provided.  |
+{:.grid}
+
+###### Response `payload`
+
+| Property              | Optionality | Type   | Description |
+| --------------------- | ----------- | ------ | ----------- |
+| `status`              | REQUIRED    | string | An HTTP response code (i.e. "200 OK" or "404 NOT FOUND"). |
+| `resources`           | OPTIONAL    | object | [FHIR Bundle] containing zero or more resources matching the requested `location`. |
+| `outcome`             | OPTIONAL    | object | [FHIR OperationOutcome] resulting from the message action. |
+{:.grid}
+
+All resources included in the response `resources` bundle SHALL contain a
+`location` value which matches the value specified in the request `location`
+field; however, if no value was provided in the request, the response bundle
+SHALL contain the *full* scratchpad contents.
```

Comment:
The phrasing on this is awkward; might say something like:

> All `Bundle.entry.response` objects SHALL include a `location` of the form `{{ResourceType}}/{{Id}}`. If the request included a `location`, the response SHALL include a single entry if the requested location exists, and zero entries otherwise (*). If the request omitted a `location`, the response SHALL an entry for each resource currently on the scratchpad.

`*` if you'd rather specify that the response is omitted, that's fine -- but I'd aim for just one "right way" to express this, whichever we pick

Reply:
PTAL - this section has drifted a ways from when this comment was first made.